src/pages/user_profile/AffiliationAndRoles.tsx

```diff
@@ -16,7 +16,9 @@ export default function AffiliationAndRole(props: AffiliationAndRoleProps) {
   useEffect(() => {
     const init = async () => {
       try {
-        const allRoles = user.roles.map((role) => role.name).join(', ');
+        // Note that although user is a required prop and roles is a required property, the parent JSX component may
+        // not honor that and pass an undefined entity during load.
```

Comment:
If the proper fix is to change this code, then shouldn't the typescript type for `user` and `roles` be updated to match this situation?

Or is there a way to fix the parent component so it initializes this data so there's no undefined data?

Reply:
I'm not sure what to do about conditionally optional properties in TS. User is a required prop, and roles is a required field on user, so the current model is correctly defined at present. I think the root problem is that we're mixing JSX and TS. Refactoring the parent component is, in my mind, a different ticket and out of scope.